documentation/specs/proposed/BuildCheck.md

```diff
@@ -0,0 +1,392 @@
+
+The analysis has small impact on build duration with ability to opt-out from analysis altogether which will remove all the performance costs associated with the analysis.
+
+
+# Scope of initial iteration
+
+Majority of following cases are included in appropriate context within the scenarios in [User Experience](#user-experience) section. Following is a quick overview.
+
+**In scope**
+* Inbox (build-in) analyzers that run during the build execution.
+* Inbox analyzers that run when replaying binlog.
+* Custom authored analyzers, delivered via nuget.
+* Analyzers reports (errors, warnings, messages) are in logger output, VS error window.
+* Codes will be distinguishable from standard build warnings/error (to prevent easy mixups and attempts to configure standard msbuild warnings/errors via editorconfig), but otherwise the outputs are very similar.
+* Default opt-ins and levels for inbox analyzers set by sdk version (via [`$SdkAnalysisLevel`]((https://github.com/dotnet/designs/blob/main/proposed/sdk-analysis-level.md))) or other agreed mechanism for controling increasing strictness between .NET versions.
+* Custom analyzers opted in via `PackageReference` of a particular nuget with the analyzer.
+* Explicit overrides of enablement and analysis levels via `.editorconfig` file (with up to a per project scope).
+* Standards of `.editorconfig`s will be observed. 
+* Simplified authoring experience via template and doc.
+* Single analyzer can produce reports for multiple rules. However those need to be declared upfront.
+* Opt-in reporting of time spent via specific analyzers and infra overall.
+* Collect touched `.editorconfg`s into binlog embedded files.
+* Possibility to opt-out from analysis - the perf should not be impacted when done so.
+* Team collects performance impact numbers on a set of benchmark builds with the inbox analyzers enabled.
+
+**Non Goals, but subject for consideration**
+* Custom anlyzer in a local project (source codes) or a binary.
+* Bulk configuration of multiple rules (based on prefixes).
+* Specifying scope of MSBuild imports that will be considered for analysis (so that e.g. data from sdk won't even be passed to anlyzer, if not requested).
+* Attempts to try to configure standard msbuild warnings/errors via `.editorconfig` should lead to fail fast errors.
+* Configuring analysis levels when analysing from binlog - beyond the collected editorconfigs
+* Rich information in VS error window.
+
+
+**Out of scope**
+* Design time build analysis.
+* Localization support (for reports message formats, identifiers, etc.).
```

Comment:
writing down that we should check on Roslyn's loc story and make sure we aren't causing problems here--MSBuild is multilingual first.

Reply:
@baronfel would you see a localization support for the custom authored analyzers as a must have requirement for a V1?

The inbox analyzers will be localized as any other MSBuild code.

Custom analyzers are currently not planned to be localizable for V1 - similarly as we lack the support for the custom tasks.

On the other hand Roslyn provides such support today:
https://github.com/dotnet/roslyn/blob/main/docs/analyzers/Localizing%20Analyzers.md